crates/provider/src/builder.rs

```diff
@@ -225,23 +225,25 @@ impl<L, N> ProviderBuilder<L, N> {
         Ok(self.on_client(client))
     }
 
+    /// Build this provider with reqwest HTTP transport.
     #[cfg(feature = "reqwest")]
     pub fn on_reqwest_http(self, url: url::Url) -> Result<L::Provider, TransportError>
     where
         L: ProviderLayer<RootProvider<N, BoxTransport>, N, BoxTransport>,
         N: Network,
     {
-        let client = ClientBuilder::default().reqwest_http(url);
+        let client = ClientBuilder::default().reqwest_http(url).boxed();
         Ok(self.on_client(client))
     }
 
+    /// Build this provider with hyper HTTP transport.
     #[cfg(feature = "hyper")]
     pub fn on_hyper_http(self, url: url::Url) -> Result<L::Provider, TransportError>
     where
         L: ProviderLayer<RootProvider<N, BoxTransport>, N, BoxTransport>,
         N: Network,
     {
-        let client = ClientBuilder::default().hyper_http(url);
+        let client = ClientBuilder::default().hyper_http(url).boxed();
```

Comment:
I don't think this should be boxed

Reply:
>@prestwich Unsure why but adding these features forced the ProviderBuilder::on_client(&self, client: RpcClient<T>) to only accept RpcClient<BoxTransport>

Yes it shouldn't.

Reply:
oh I just saw this in #379 and was confused about it 🤔

Reply:
i think what happened is that the blocks were previously entirely non-functional, and therefore were written incorrectly and we didn't notice (because they were non-functional and never checked by the compiler). Now that the cfg blocks are effective, a compiler warning appears. I'll fix in #379 in a second